### ISSUE TITLE
No support for sparse target vectors in BiLSTM and CNN

### ISSUE THREAD

**Issue by nsorros** [wellcometrust/WellcomeML#138]
Our BiLSTM and CNN architecture do not work with a sparse target vector Y. This is relevant especially for MeSH tagging where the Y vector in its dense form is approx 200GB. Even though tensorflow seems to support sparse multiclass classification with sparse categorical cross entropy loss, something similar does not seem to exist for multilabel.

One way around the problem is to stream the data and densify only during the batch used which is what I have been using in my experiments with these architectures.
